internal/compute/ansible.go

```diff
@@ -28,26 +49,123 @@ type AnsibleConfigurator struct {
 	sshKeyPath string
 	// mutex protects the instances map
 	mutex sync.Mutex
+	// instanceRepo provides access to instance data
+	instanceRepo InstanceRepository
```

Comment:
I would directly pull in the Instance Service, not redefine an interface for the instance repository.